torch/testing/_internal/common_utils.py

```diff
@@ -1973,10 +1973,12 @@ def make_tensor(size, device: torch.device, dtype: torch.dtype, *, low=None, hig
     """ Creates a random tensor with the given size, device and dtype.
 
         By default, the tensor's values are in the range [-9, 9] for most dtypes. If low
-        and/or high are specified then the values will be in the range [max(-9, low), min(9, high)].
+        and/or high are specified then the values will be in the range [low, high). If low
+        and/or high are passed as -inf/inf (respectively), then the values will be minimum
+        and maximum limit of the respective dtype.
```

Comment:
Not sure about this. I'm guessing this is only for BC? Could you give an example for such a case? If we actually want `inf` / `-inf` as part of the tensor, how would we achieve that?

Reply:
> I'm guessing this is only for BC? Could you give an example for such a case? 

The use case I encountered was from the OpInfos: there are ops which have domain listed as `(value, float('inf'))` (e.g: `acosh`: https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_methods_invocations.py#L4837). Though, it's expected that these domain values will be half open, which means `float('inf')` is going to be excluded, but still if you pass it in a function like `torch.distributions.uniform.Uniform` - it will give you infinite values:

```python
>>> torch.distributions.uniform.Uniform(low=2, high=float('inf')).sample((1, 2))
>>> tensor([[inf, inf]])
```

I'm not sure if it's a BC, since earlier it used to be: https://github.com/pytorch/pytorch/blob/2e49c5dc37bce5ce8ba463a59aacb5d3e4a638b6/torch/testing/_internal/common_utils.py#L1942:L1943. It's similar to what we are doing, except we use the dtype limits and not the default low and high values.

> If we actually want inf / -inf as part of the tensor, how would we achieve that?

From what I remember from our offline discussion, this does seem to be an interesting use-case where the user might want to include interesting values like nans/inf/-inf, but passing this as `low/high` doesn't seem to be the right way. We should discuss in the sync meeting, about having `include_inf` / `include_nan` like arguments OR `include_values` and allow users to specify the values they want to be included. Hopefully, these changes can be discussed and then implemented in the follow-up PRs.

Thanks for raising the question, hope this answers well. Let me know if you have any follow-up questions. @pmeier

Reply:
Going along with the `acosh` example, wouldn't it be better to change the `OpInfo` to `domain=(1, None)`?

Reply:
Thinking of the reason why was domain set as `domain=(val, float('inf'))` for such ops (note that there are around 10 ops with such domains in the OpInfos):

A lot of mathematical resources mention these domains as `(val, inf)` or `(-inf, val)` and if this has to go to a user-facing API, passing `float('inf')` is still intuitive compared to passing `None`.

But I agree, that this can lead to some confusion, specially if we have something like `include_inf` arg and:

```python
# This is confusing since we pass inf to high but also ask to include_inf
make_tenosr(low=2, high=float('inf'), device, dtype, include_inf=True)

# This is not confusing
make_tensor(low=2, high=None, device, dtype, include_inf=True)
```

I agree that replacing the usages in the internals from `float('inf')` to `None` wheever reasonable, is a good idea but the first point about mathematical references mentioning `float('inf')` in the domains, is still an open question to be answered. Looking forward to hear your opinion. Thanks!